### DISCUSSION TITLE
How can I mute the stream?

### DISCUSSION THREAD

**Discussion by HaikuOezu** [shogo4405/HaishinKit.kt#103] [Q&A] [answered]
I'm trying to implement a simple mute toggle in an app that streams via RTMP, on HaishinKit Swift all I do is set muted to true in audioSettings but Haishinkit.kt has no such feature so I tried the following.

1) attachAudio(null)
This freezes the stream

2) setting the stream's AudioRecordSource muted property to true
This causes a java.lang.UnsupportedOperationException at com.haishinkit.media.AudioRecordSource.read(AudioRecordSource.kt:129) and at com.haishinkit.media.AudioRecordSource.onInputBufferAvailable(AudioRecordSource.kt:116)
This happens even if the audiosource is muted before even streaming

I'm completely at a loss on what to do here...

**Comment by shogo4405** [comment, marked as answer]
Thank you for your quetion.
- This is a bug. I fixed this PR https://github.com/shogo4405/HaishinKit.kt/pull/104.
  - You can use `rtmpStream.audioSettings.muted = true`.

**Reply by HaikuOezu** [reply]
Thank you!! 🙏
